src/f32/funcs.rs

```diff
@@ -139,6 +197,64 @@ pub(crate) mod sse2 {
         _mm_or_ps(result, _mm_castsi128_ps(test))
     }
 
+    /// Returns a vector whose components are the corresponding components of Angles modulo 2PI.
+    #[inline]
+    pub(crate) unsafe fn m128_mod_angles(angles: __m128) -> __m128 {
+        // From DirectXMath: XMVectorModAngles
+        let v = _mm_mul_ps(angles, PS_RECIPROCAL_TWO_PI.m128);
+        let v = m128_round(v);
+        m128_neg_mul_sub!(PS_TWO_PI.m128, v, angles)
+    }
+
+    /// Computes the sine of the angle in each lane of `v`. Values outside
+    /// the bounds of PI may produce an increasing error as the input angle
+    /// drifts from `[-PI, PI]`.
+    #[inline]
+    pub(crate) unsafe fn m128_sin(v: __m128) -> __m128 {
+        // From DirectXMath: XMVectorSin
+        //
+        // 11-degree minimax approximation
+        //
+        // Force the value within the bounds of pi
+        let mut x = m128_mod_angles(v);
```

Comment:
Now that you are using mod angles again, is the accuracy OK with values outside of +/- PI? Just wondering if the function comment is still accurate?

Reply:
That depends on your definition of "OK". It seems that the error increases as you move further away from `[-PI, PI]` in either direction, but its still functions as a working `sin`. The output was rubbish outside the bounds of `PI` without `mod_angles`.